### PR TITLE
Internal API endpoint to support PaC add pipeline page

### PR DESCRIPTION
**Pull request by marques-work**
`/api/admin/internal/pac/*`:

- `/preview/:plugin_id`: Exports arbitrary pipeline config JSON as plugin-supported format
    * Will very likely need to add full config validation option enabled via query param.

### CODE REVIEW

File: api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+import java.util.function.Consumer;
+
+import static com.thoughtworks.go.api.util.HaltApiResponses.haltBecauseOfReason;
+import static com.thoughtworks.go.spark.Routes.PaC.BASE_INTERNAL_API;
+import static com.thoughtworks.go.spark.Routes.PaC.PREVIEW;
+import static java.lang.String.format;
+import static spark.Spark.*;
+
+@Component
+public class PipelinesAsCodeMiscControllerV1 extends ApiController implements SparkSpringController {
```
Comment (ibnc) [inline api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java]: I don't love naming a class miscellaneous, but it's a nitpick. I feel like `PipelinesAsCodeInternalControllerV1` would be fine.

It's 100% up to you. Changing the name or not doesn't affect the merge-ability of this PR.

File: api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        }
+
+        ConfigHelperOptions options = new ConfigHelperOptions(goConfigService.getCurrentConfig(), passwordDeserializer);
+        JsonReader jsonReader = GsonTransformer.getInstance().jsonReaderFrom(req.body());
+
+        validateNamePresent(jsonReader);
+
+        PipelineConfig pipeline = PipelineConfigRepresenter.fromJSON(jsonReader, options);
+
+        if ("true".equalsIgnoreCase(req.queryParams("validate"))) {
```
[the remark this answers is not in the document]
Comment (marques-work) [reply api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java]: @ibnc that's a fair point.

File: api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            if (!performFullValidate(pipeline, groupName)) {
```
[the remark this answers is not in the document]
Comment (marques-work) [reply api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java]: ditto here.

File: api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java (hunk 1 above)
Comment (marques-work) [reply api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java]: Yeah, that's probably better to be explicit that this is not public.

File: api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            ExportedConfig export = repoPlugin.pipelineExport(pipeline, groupName);
+
+            res.header("Content-Type", export.getContentType());
+            res.header("Content-Disposition", format("attachment; filename=\"%s\"", export.getFilename()));
+            return export.getContent();
+        }
+
+    }
+
+    private boolean performFullValidate(final PipelineConfig pipeline, final String group) throws HaltException {
```
[the remark this answers is not in the document]
Comment (marques-work) [reply api/api-pipelines-as-code-misc-v1/src/main/java/com/thoughtworks/go/apiv1/pipelinesascodemisc/PipelinesAsCodeMiscControllerV1.java]: I'll change it as it does return a `boolean` for a check, as you said.

**Pull request merged**